Slim/Formats/XML.pm

```diff
@@ -416,8 +416,18 @@ sub parseRSS {
 		
 		$feed{'image'} = $url;
 	}
-	elsif ( $xml->{'itunes:image'} ) {
-		$feed{'image'} = $xml->{'itunes:image'}->{'href'};
+	elsif ( ref $xml->{'channel'}->{'itunes:image'} ) {
+		$feed{'image'} = $xml->{'channel'}->{'itunes:image'}->{'href'};
+	}
+
+	# Jive browsing (SlimBrowserApplet) will break when fetching artwork
+	# unless it receives its icon url as a scalar value. So ensure that we
+	# really *do* have a scalar.
+	# E.g. a broken podcast that provides an empty 'url' tag is known to
+	# trigger this unwelcome behaviour.
+	if (ref $feed{'image'})	{
```

Comment:
I think we treat this case in other places and would just pick the first of a list, in case `image` was a list. Could you please incorporate this?

Reply:
I've revised the approach in line with your suggestion.
As I usually muck up revisions to these things, I've prepare a new branch here:
https://github.com/mw9/slimserver/tree/feature/podcast_issues_revised

But I've not had a chance to check them in running code yet...

If you're happy with the revised approach, and they do run as expected, I propose that I then push  up the revised  branch. Make sense ?

Reply:
I've revised the changes to take up what I think you're after.

In the process I've added a "tidy up" commit to the "Lage der Nation" issue where we had a list of images to deal with. I did this because I became confused. I have checked that the tidied up version stills works  as expected.

All other changes seem to be working fine.

So you should be seeing four replacement (forced) commits, added today.

I hope I haven't confused matters...